lib/hooks/core/hook-http.js

```diff
@@ -54,19 +53,21 @@ function requestWrap(request) {
       return request.apply(this, arguments);
     }
 
+    var localOptions = (typeof options === 'string') ?
```

Comment:
Use the same check as above, i.e. `typeof options === 'object'`. That way the optimizer (value numbering) can realize that it is the same check as before, and void the second check completely.